src/features/orders/OrderDetail.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback } from 'react'
+import { actions, detailSelectors } from './reducer'
+import Drawer from '@material-ui/core/Drawer'
+import Box from '@material-ui/core/Box'
+import { makeStyles, Theme } from '@material-ui/core/styles'
+import Grid from '@material-ui/core/Grid'
+import Icon from '@material-ui/core/Icon'
+import IconButton from '@material-ui/core/IconButton'
+import CloseIcon from '@material-ui/icons/Close'
+import Typography from '@material-ui/core/Typography'
+import OrderIcon from '@material-ui/icons/ShoppingCart'
+import OrderTabPanel from './OrderTabPanel'
+import { useSelector, useDispatch } from 'react-redux'
+
+const useStyles = makeStyles((theme: Theme) => ({
+    root: {
+        maxWidth: '90vw',
+        width: '40em',
+    },
+    detailIdWrapper: {
+        marginLeft: '1em',
+    },
+    background: {
+        backgroundColor: theme.palette.background.default,
+    }
+}));
+
+export default function OrderDetail() {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+
+    const isOpen = useSelector(detailSelectors.isOpen);
+    const order = useSelector(detailSelectors.currentOrder);
+    const tab = useSelector(detailSelectors.tab);
+    const id = useSelector(detailSelectors.id);
+
+    const handleClose = useCallback(
+        () => dispatch(actions.detail.close()), 
+        [dispatch],
+    );
+    const handleChangeTab = useCallback(
+        (tabId: number) => dispatch(actions.detail.changeTab(tabId)),
+        [dispatch],
+    );
+
+    return (
```

Comment:
please split this component into smaller local components (located in this file). For instance I'd extract Header and Title components. Now markup looks too 'deep'